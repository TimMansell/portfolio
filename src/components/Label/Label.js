import React from 'react';
import PropTypes from 'prop-types';

import styles from './Label.module.scss';

export const Label = ({ label }) => {
  return <div className={styles.label}>{label}</div>;
};

Label.propTypes = {
  label: PropTypes.string.isRequired,
};

export default Label;
