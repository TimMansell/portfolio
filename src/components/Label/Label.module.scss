@import 'variables';
@import 'utilities/pxtorem';

.label {
  color: var(--colour-5);
  padding: var(--vr-xxs);
  border-radius: 3px;
  font-size: pxtorem(12);
  margin-bottom: var(--vr-xxs);
}

.primary {
  background: var(--colour-1);
}

.secondary {
  background: var(--colour-3);
}
