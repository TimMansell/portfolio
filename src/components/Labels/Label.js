import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import styles from './Label.module.scss';

export const Label = ({ label, type, size }) => {
  const labelClasses = classnames(styles.label, {
    [styles.isPrimary]: type === 'primary',
    [styles.isSecondary]: type === 'secondary',
    [styles.isTertiary]: type === 'tertiary',
    [styles.isLarge]: size === 'lg',
  });

  return <li className={labelClasses}>{label}</li>;
};

Label.propTypes = {
  label: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['primary', 'secondary', 'tertiary']),
  size: PropTypes.oneOf(['lg']),
};

export default Label;
