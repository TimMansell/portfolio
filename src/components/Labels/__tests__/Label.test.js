import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Label from '../Label';

const defaultProps = {
  label: 'test label',
  type: 'primary',
};

describe('Label', () => {
  it('should render my component', () => {
    const props = {
      ...defaultProps,
    };

    render(<Label {...props} />);
  });

  it('should match snapshot', () => {
    const props = {
      ...defaultProps,
    };
    const { asFragment } = render(<Label {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should display a primary background', () => {
    const props = {
      ...defaultProps,
    };
    const { getByText } = render(<Label {...props} />);

    expect(getByText(defaultProps.label)).toHaveClass('isPrimary');
  });

  it('should display a secondary background', () => {
    const props = {
      ...defaultProps,
      type: 'secondary',
    };
    const { getByText } = render(<Label {...props} />);

    expect(getByText(defaultProps.label)).toHaveClass('isSecondary');
  });

  it('should display a tertiary background', () => {
    const props = {
      ...defaultProps,
      type: 'tertiary',
    };
    const { getByText } = render(<Label {...props} />);

    expect(getByText(defaultProps.label)).toHaveClass('isTertiary');
  });

  it('should render as large label', () => {
    const props = {
      ...defaultProps,
      size: 'lg',
    };
    const { getByText } = render(<Label {...props} />);

    expect(getByText(defaultProps.label)).toHaveClass('isLarge');
  });
});
