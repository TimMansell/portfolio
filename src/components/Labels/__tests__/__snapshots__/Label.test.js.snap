// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should match snapshot 1`] = `
<li
  className="label isPrimary"
>
  test label
</li>
`;
